### ISSUE TITLE
Frontmatter YAML editor

### ISSUE THREAD

**Issue by timotoots** [nextcloud/text#3571]
**Is your feature request related to a problem? Please describe.**
The latest update is great that you can edit frontmatter YAML fields. Thanks! But it's a raw text only.

**Describe the solution you'd like**
 It would be great if it gets developed into more user-friendly YAML editor. So the fields could be added and removed more like a table interface.

**Describe alternatives you've considered**
It's possible to edit now the fields, but to give front matter in hands of non-technical personnel is complicated.

**Additional context**

**Comment by susnux** [comment]
As I personally use it for editing files for `hugo`, I would love to have this.

But if someone is about to develop this, please keep in mint, that front matters can have any format.
They are not restricted to `yaml`, I already have seen `json` and even `toml`.

**Comment by max-nextcloud** [comment]
I think one challenge here also is that all these front matter file formats allow defining nested key value mappings at an arbitrary depth such as
```yaml
person:
    name:   Dorothy
    parents: 
        mother:
            name: Helen
            parents:
               mother:
```

I guess you get my point. These languages are all so complex that writing a generic editor for them is complex as well and might not result in an intuitive interface either.

Given the [examples on hugos homepage](https://gohugo.io/content-management/front-matter/#example) typical frontmatter seems to be far less complex. I think the main problem is to come up with a way to easily edit this simple subset without breaking more complex structures.

**Comment by timotoots** [comment]
Yes, I also started using Text for Hugo files, before it was not possible. It make it great to collaborate on static website editing.

I think supporting two levels of nesting would cover 95% of the use cases. Even without nested fields it would be helpful, because simple pages just use title and description fields.
